fastplotlib/layouts/_gridplot.py

```diff
@@ -399,3 +405,12 @@ def update_current_subplot(self, ev):
                 self.panzoom_controller_button.value = subplot.controller.enabled
                 self.maintain_aspect_button.value = subplot.camera.maintain_aspect
 
+    def record_plot(self, obj):
+        if self.record_button.value:
+            try:
+                self.plot.record_start(f"./{datetime.now().isoformat()}.mp4")
```

Comment:
ISO format returns colons and periods, so replace those with underscores. I think windows will complain with those characters. I think you can just do `.replace()` on a str